pkg/sql/schemachanger/schemachanger_test: TestRollback_drop_column_unique_index failed

pkg/sql/schemachanger/schemachanger_test.TestRollback_drop_column_unique_index [failed](https://tanzanite.cluster.engflow.com/invocations/default/0fc06040-1c14-429f-9c19-5b1ce54409d5?testReportRun=2&testReportShard=16&testReportAttempt=1#targets-Ly9wa2cvc3FsL3NjaGVtYWNoYW5nZXI6c2NoZW1hY2hhbmdlcl90ZXN0) on master @ [e999ede1ed2bc87b34e01eedba1da58e09b8c4d3](https://github.com/cockroachdb/cockroach/commits/e999ede1ed2bc87b34e01eedba1da58e09b8c4d3):

Fatal error:

```
panic: test timed out after 39m57s
running tests:
	TestRollback_drop_column_unique_index (2m33s)
	TestRollback_drop_column_unique_index/post_commit_stage_6_of_7 (3s)

```
Stack: 

```
goroutine 196064061 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x385
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestRollback_drop_column_unique_index
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestRollback_drop_column_unique_index2302162674
    test_log_scope.go:81: use -show-logs to present logs inline
    datadriven.go:144: 
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/sql/schemachanger/schemachanger_test_/schemachanger_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/schemachanger/testdata/end_to_end/drop_column_unique_index/drop_column_unique_index.definition:1:
        setup [0 args]
        CREATE DATABASE t;
        CREATE TABLE t.test (k INT PRIMARY KEY, v INT, pi DECIMAL DEFAULT (DECIMAL '3.14'));
        CREATE UNIQUE INDEX vidx ON t.test (v);
        ALTER TABLE t.test ADD COLUMN x DECIMAL DEFAULT (DECIMAL '1.4') CHECK (x >= 0);
        ----
=== RUN   TestRollback_drop_column_unique_index/post_commit_stage_6_of_7
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=16</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRollback_drop_column_unique_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39031

--- a/pkg/sql/schemachanger/BUILD.bazel
+++ b/pkg/sql/schemachanger/BUILD.bazel
@@ -38,7 +38,7 @@ go_test(
         "//build/toolchains:is_heavy": {"test.Pool": "heavy"},
         "//conditions:default": {"test.Pool": "large"},
     }),
-    shard_count = 16,
+    shard_count = 32,
     deps = [
         "//pkg/base",
         "//pkg/build/bazel",

